mongo/datadog_checks/mongo/mongo.py

```diff
@@ -76,6 +76,9 @@ def __init__(self, name, init_config, instances=None):
 
     @property
     def api_client(self):
+        # This needs to be a property for our unit test mocks to work.
+        if self._api_client is None:
+            self._api_client = MongoApi(self._config, self.log)
         return self._api_client
```

Comment:
```suggestion
    @cached_property
    def api_client(self):
        # This needs to be a property for our unit test mocks to work.
        return MongoApi(self._config, self.log)
```

maybe?

Reply:
I'll try this out locally to avoid git churn!